### ISSUE TITLE
How to save Int64 interger for PHP

### ISSUE THREAD

**Issue by shohanshohan** [mongodb/mongo-php-library#590]
also new \MongoInt64() can 
but now it can not 
just save int32   
how to save as 1541753675000 into int64 not double or object
thank you

**Comment by shohanshohan** [comment]
MongoDB\BSON\Int64   
it is private 
why

**Comment by derickr** [comment]
The MongoDB driver will translate any integer in a PHP variable (for example ``$var = 1541753675000``) into an appropriate BSON type.

On 64-bit platforms, if the number fits in the 32-bit signed ranged, the driver will use a BSON int32 type, and if the number does not fit in the 32-bit signed range, it will use a BSON int64 type instead.
When converting data from the database, both types will be translated into a PHP integer value.

On 32-bit platforms, PHP can not represent integer numbers that do not fit in the 32-bit signed range. On these platforms, the following already shows a double:

```
$ /usr/local/php/7.2dev-32bit/bin/php -r "var_dump(1541753675000);"
float(1541753675000)
```
As PHP already sees this as a double, the driver will translate this PHP variable into a BSON double in the database, as that's the appropriate type. PHP integers will be stored as a 32-bit BSON integer in the database.
When reading data from the database, the BSON int32 type will be translated into a normal PHP integer, and a BSON double type will be translated into a PHP double/floating point number.
However, as PHP integer on 32-bit platforms *cannot* hold the contents of a BSON int64 type, the driver wraps this value in a ``\MongoDB\BSON\Int64`` class so that it can preserve its value. As there is no safe way to handle 64-bit integers on 32-bit platforms, the only way to make sure that you can round-trip the value (read from the database, and then store it again) is to use this wrapper class.

In short, if you need to deal with 64-bit integers, you need to make sure you're on a 64-bit platform, and everything works as expected.
